src/activity_handlers.cpp

```diff
@@ -2546,7 +2546,11 @@ struct weldrig_hack {
 
     item &get_item() {
         if( veh != nullptr && part >= 0 ) {
-            pseudo.charges = veh->drain( itype_battery, 1000 - pseudo.charges );
+            item pseudo_magazine( pseudo.magazine_default() );
+            pseudo.put_in( pseudo_magazine, item_pocket::pocket_type::MAGAZINE_WELL );
+            pseudo_magazine.contents.clear_items();
```

Comment:
This statement appears to be pointless. It modifies the variable `pseudo_magazine` (well, one of its members), but that should not have any effect on other variables. It would make sense *before* the call to `pseudo.put_in`.